aws/credentials.go

```diff
@@ -125,6 +175,13 @@ type CredentialsProvider interface {
 	Retrieve(ctx context.Context) (Credentials, error)
 }
 
+// CredentialProviderChain allows any credential provider to track
+// the chain where they were sourced. For example, if the credentials came from a
+// call to a role specified in the profile, this method will give the whole precedence chain
+type CredentialProviderChain interface {
```

Comment:
I don't think we can name it like this - `ProviderChain` really implies that this is a provider _composed_ of multiple providers, i.e. a chain that tries providers until one resolves.

Reply:
Yeah agree. Discussed offline and changed